doc/example/example_nonlinear_monotone.ipynb

```diff
@@ -0,0 +1,856 @@
+       "16     4.913791     5                   NaN                1   \n",
+       "17     4.929008     5                   NaN                1   \n",
+       "\n",
+       "   observableTransformation noiseDistribution     measurementType  \n",
+       "0                       lin            normal  nonlinear_monotone  \n",
+       "1                       lin            normal  nonlinear_monotone  \n",
+       "2                       lin            normal  nonlinear_monotone  \n",
+       "3                       lin            normal  nonlinear_monotone  \n",
+       "4                       lin            normal  nonlinear_monotone  \n",
+       "5                       lin            normal  nonlinear_monotone  \n",
+       "6                       lin            normal  nonlinear_monotone  \n",
+       "7                       lin            normal  nonlinear_monotone  \n",
+       "8                       lin            normal  nonlinear_monotone  \n",
+       "9                       lin            normal  nonlinear_monotone  \n",
+       "10                      lin            normal  nonlinear_monotone  \n",
+       "11                      lin            normal  nonlinear_monotone  \n",
+       "12                      lin            normal  nonlinear_monotone  \n",
+       "13                      lin            normal  nonlinear_monotone  \n",
+       "14                      lin            normal  nonlinear_monotone  \n",
+       "15                      lin            normal  nonlinear_monotone  \n",
+       "16                      lin            normal  nonlinear_monotone  \n",
+       "17                      lin            normal  nonlinear_monotone  "
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from pandas import option_context\n",
+    "\n",
+    "with option_context('display.max_colwidth', 400):\n",
+    "    display(petab_problem.measurement_df)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Constructing the objective and pypesto problem"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Different options can be used for the spline approximations:\n",
+    "\n",
+    "- `spline_ratio`: float value, determines the number of spline knots as `n_spline_pars` = `spline_ratio` * `n_datapoints`  \n",
+    "- `use_minimal_difference` : `True` / `False`\n",
+    "\n",
+    "The `use_minimal_difference` option determines whether a minimal difference between the spline heights will be constrained. For most models, enabling this option will increase convergence by increasing error of incorrect ordering. Nevertheless, the option has to be used with caution, as it will reduce the spline's capability of approximating functions with flat regions well. This we will show in this notebook."
```

Comment:
> increase convergence by increasing error

?

Reply:
Pretty bad wording. I've rephrased it:

Enabling this option can improve convergence for most models by increasing the objective function value for incorrect orderings. However, the option has to be used with caution as it may reduce the spline's ability to approximate functions with flat regions accurately. This will be demonstrated in the last section.

Reply:
Changed again due to new implementation of minimal difference:

The `min_diff_factor` option determines to which degree the spline heights will be constrained to have a minimal difference. Setting a non-zero value can improve convergence for most models by increasing the objective function value for incorrect orderings. However, high values of the `min_diff_factor` have to be used with caution as they will reduce the spline's ability to approximate functions with flat regions accurately. This will be demonstrated in the last section.

Reply:
> The `min_diff_factor` is a multiplier of the minimal difference between spline heights. Positive values act as a penalty in the objective function for incorrect orderings; this penalty can improve convergence for most models. However, high `min_diff_factor` values will reduce the spline's ability to approximate functions with flat regions accurately. This issue will be demonstrated in the last section.

Reply:
Much nicer, thank you.